File: perl/dog/test_dog.pl
#!/usr/bin/perl
use warnings;
use strict;
use dog;
sub main {
    my $my_dog = dog->new("Lily", "Pit Mix", 8);
    print $my_dog . "\n";
    print $my_dog->info . "\n";
    $my_dog->birthday;
    print $my_dog->info . "\n";
    print $my_dog->greet("hello");
}
main();

File: java/Dog/test_dog.java
import java.io.*;
import java.util.*;
public class test_dog {
  public static void main (String[] args) {
     Dog my_dog = new Dog("Happy", "Beagle", 6);
     System.out.println(my_dog.greet("Hello"));
     System.out.println(my_dog);
     my_dog.birthday();
     System.out.println(my_dog);
     System.out.println(my_dog.greet("Bye"));
  }
}